### ISSUE TITLE
Build failure: nvidia-x11

### ISSUE THREAD

**Issue by Reikott** [NixOS/nixpkgs#357643]
## Steps To Reproduce

Run nix flake update and nixos-rebuild switch --flake to update to the latest nixos-unstable. Nvidia fails to build.

## Additional context

![NvidiaBuildFailure](https://github.com/user-attachments/assets/b99e0df7-02cd-4240-a8b5-ab5901c06e4d)

## Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.8, NixOS, 25.05 (Warbler), 25.05.20241115.5e4fbfb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `""`
 - nixpkgs: `/nix/store/hfz1qqd0z8amlgn8qwich1dvkmldik36-source`

**Comment by Vilsol** [comment]
Same issue here. On `nixos-unstable-small` flake

 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.5-zen1, NixOS, 24.11 (Vicuna), 24.11.20241107.85f7e66`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(vilsol): `""`
 - channels(root): `"nixos-23.11, nixos-unstable, nixpkgs-unstable"`
 - nixpkgs: `/nix/store/gg86rfp39vc7chqsszk32q7995hz4943-source`

**Comment by VeilSilence** [comment]
for 6.12 kernel [patch](https://github.com/Binary-Eater/open-gpu-kernel-modules/commit/8ac26d3c66ea88b0f80504bdd1e907658b41609d.patch) is [required](https://forums.developer.nvidia.com/t/patch-for-565-57-01-linux-kernel-6-12/313260)

**Comment by Reikott** [comment]
The drivers seem to work if I'm using the beta. But there are occasional visual artifacts.

**Comment by pbek** [comment]
Sticking to Kernel 6.11 seems to be a workaround for now.

**Comment by UlyssesZh** [comment]
Both the commenters that posted their environments use kernel 6.11, not?

**Comment by pbek** [comment]
> Both the commenters that posted their environments use kernel 6.11, not?

Yes, as a workaround until the package works with 6.12.

**Comment by UlyssesZh** [comment]
Oh, sorry, I misunderstood. I thought 6.11 is the kernel with which they are trying to build nvidia-x11 against.

**Comment by leonm1** [comment]
Fixed for the beta version in #358047. To use the beta version, the following config is required:

```nix
hardware.graphics.package = config.boot.kernelPackages.nvidiaPackages.beta;
```

**Comment by Reikott** [comment]
Currently using the beta. Works fine. Although there are still occasional graphical glitches.

Although not sure if those glitches are related to Nvidia or KDE.

**Comment by AchmadFathoni** [comment]
> Fixed for the beta version in #358047. To use the beta version, the following config is required:
> 
> ```
> hardware.graphics.nvidia.package = config.boot.kernelPackages.nvidiaPackages.beta;
> ```

This PR is merged in nixos-24.11. But it still marked as broken when I use the latest kernel `pkgs.linuxPackages_latest`
```
error: Package ‘nvidia-x11-560.35.03-6.12.3’ in /nix/store/mxixd08qh56q3ksmxns23id8n14r0gvj-source/pkgs/os-specific/linux/nvidia-x11/generic.nix:278 is marked as broken, refusing to evaluate
```

**Comment by PassiveLemon** [comment]
No longer getting this error, appears to be fixed?
